using_restricted_array.rb

```diff
@@ -60,7 +134,29 @@ def find_largest(array, length)
 # (Hint: if there are elements with 'SPECIAL_VALUE', there is no room to insert)
 # All subsequent elements will need to be moved forward by one index.
 def insert_ascending(array, length, value_to_insert)
-  puts "NOT IMPLEMENTED"
+  i = 0
+
+  until array[i] > value_to_insert
+    if array[i] == SPECIAL_VALUE
+      return
+    end
+    i += 1
+  end
+
+  insert_index = i
+  temp = array[i]
+  next_val = array[i + 1]
+
+  until i == length - 1
+    if array[i] == SPECIAL_VALUE
+       return
+    end
+  array[i + 1] = temp
+  i += 1
+  temp = next_val
+  next_val = array[i + 1]
+
+  end

```

Comment:
At some point, you should update array[insert_index] to have the value of value_to_insert before returning.